bokehjs/src/coffee/models/widgets/data_table.coffee

```diff
@@ -99,6 +99,10 @@ export class DataTableView extends WidgetView
     @grid.invalidate()
     @grid.render()
 
+    # This is only needed to call @_tell_document_about_change()
+    @model.source.data = @model.source.data
+    @model.source.change.emit()
```

Comment:
@DuCorey , @bryevdv This doesn't propagate the update back to Python's `source.data`, does it?

Reply:
It does propagate changes back to Python. This happens unconditionally (almost, there's `silent` option) when a property is assigned to. Additional `emit` is necessary to trigger internal bokehjs' signals, because this requires `!isEqual(old_data, new_data)`, which isn't the case here.

Reply:
@mattpap I'm almost certain that I saw `change.emit()` propagate changes on the js side, but not back to Python side... WOuld you have a counterexample?

Reply:
> I'm almost certain that I saw change.emit() propagate changes on the js side, but not back to Python side.

That's true, but `emit` and propagation of changes to Python are two independent features. To propagate changes to Python it's sufficient to assign to a property. If you look at https://github.com/bokeh/bokeh/blob/master/bokehjs/src/coffee/core/has_props.ts#L254-L283, knowing that `obj.prop = val` translates to `obj.setv({prop: val})`, then it should be clear that changes are propagated as long as `silent` is not set or is `false`.